cmd/wego/app/add/cmd.go

```diff
@@ -113,24 +113,28 @@ func runCmd(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	var providerName gitproviders.GitProviderName
+	var normalizedUrl gitproviders.NormalizedRepoURL
 	// We re-use the same --url flag for both git and helm sources.
 	// There isn't really a concept of "provider" in helm charts, and there is nothing to push.
 	// Assume charts are always public and no auth needs to be done.
 	if !isHelmRepository {
-		providerName, err = gitproviders.DetectGitProviderFromUrl(repoUrlString)
+		normalizedUrl, err = gitproviders.NewNormalizedRepoURL(repoUrlString)
 		if err != nil {
-			return fmt.Errorf("error detecting git provider: %w", err)
+			return fmt.Errorf("error creating normalized url: %w", err)
 		}
+		params.GitProviderName = normalizedUrl.Provider()
+	} else {
+		// helm doesnt use the gitprovider but it still needs to be set to something for now
+		params.GitProviderName = gitproviders.GitProviderGitHub
```

Comment:
@J-Thompson12 How would you feel about a `gitproviders.GitProviderNone` or something like that. I would also be okay with setting this to some zero value.

Do you have a sense for where this is required? Is it only in the `auth.NewAuthCLIHandler` or is it other places?

Reply:
its being used here buildGitProvider() to create the git provider. I can certainly find a way to skip all of that but the current behavior is to look for GITHUB_TOKEN even if its helm. i am pretty sure all this behavior will change soon with @jrryjcksn changes.

Reply:
Ok nevermind. We can leave Github as the default then.